src/make-request.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import type {SuperAgentRequest} from "superagent";
+import type {RenderGatewayOptions} from "./types.js";
+import {isCacheable} from "./is-cacheable.js";
+import type {Logger} from "./shared/index.js";
+import {makeUnbufferedNoCacheRequest} from "./make-unbuffered-no-cache-request.js";
+import {asCachedRequest, asUncachedRequest} from "./requests-from-cache.js";
+
+/**
+ * Make a request for a given URL
+ *
+ * Could resolve from cache if caching is enabled and the request has already
+ * been fulfilled once. Otherwise, this creates a new request for the URL.
+ *
+ * The request will resolve with an additional _fromCache property, which will
+ * indicate if it was resolved from cache or not.
+ *
+ * @param {RenderGatewayOptions} options The options used to start the gateway.
+ * @param {string} url The URL to be requested.
+ * @param {Logger} logger The logger to use.
+ * @param {boolean} [buffer] Defaults to true. When true, the response body will
+ * be buffered, otherwise it will not.
+ * @returns {SuperAgentRequest} A superagent request for the URL.
+ */
+export const makeRequest = (
+    options: RenderGatewayOptions,
+    url: string,
+    logger: Logger,
+    buffer?: boolean = true,
+): SuperAgentRequest => {
```

Comment:
To actually make the request we'd do the following right:
```
const req = makeRequest(...);
...
req.end();
```
Sorry for asking newbie questions but I'm unfamiliar with backend stuff. 😞

Reply:
That depends on whether `buffer` is true or not.

If it is true, we can just `await` the request (that's the usage we'll be using). When it isn't true, we might need to call `end`. You can checkout the superagent docs on that though.

Our usage is likely to be with `buffer` as true since that's what we do in RRS, but I wanted to add the flexibilty here as we might use this API for more requests than RRS does and I wasn't sure that all cases would like being forced to `buffer`.

Reply:
It seems like that would be some methods like `set` would need to be called in the same event loop cycle as when the request was made while others like `prune` or `buffer` might be callable later (though it's probably not a good idea to do so).

Reply:
It's not quite like that. The easiest analogy though perhaps inaccurate its to testing where describe and it and before all execute to register the operations and then the test state machine runs to execute those operations in a defined order.

Calls to buffer or end are what kick that state machine off.

Reply:
I didn't realize `buffer` was similar to `end`.  That makes much more sense than what I was thinking.

Reply:
Yeah, I think the `superagent` docs call out that you need to use one or the other.